src/rpcrawtransaction.cpp

```diff
@@ -29,43 +29,106 @@ extern Upgrader upgrader;
 extern Checker checker;
 #endif
 
-std::string GetTxProject(uint256 hash, int& out_blocknumber, int& out_blocktype, double& out_rac)
+void GetTxStakeBoincHashInfo(json_spirit::mObject& res, const CMerkleTx& mtx)
 {
-    CTransaction tx;
-    uint256 hashBlock = 0;
-    std::string error_code = "";
+    assert(mtx.IsCoinStake() || mtx.IsCoinBase());
 
+    res["blockhash"]=mtx.hashBlock.GetHex();
 
-    if (!GetTransaction(hash, tx, hashBlock))
+    // Fetch BlockIndex for tx block
+    CBlockIndex* pindex = NULL;
+    CBlock block;
     {
-        return "";
+        map<uint256, CBlockIndex*>::iterator mi = mapBlockIndex.find(mtx.hashBlock);
+        if (mi == mapBlockIndex.end())
+        {
+            res["error"] = "block_not_in_index";
+            return;
+        }
+        pindex = (*mi).second;
+        if (!block.ReadFromDisk(pindex))
+        {
+            res["error"] = "block_read_failed";
+            return;
+        }
     }
-    CDataStream ssTx(SER_NETWORK, PROTOCOL_VERSION);
-    ssTx << tx;
-    string strHex = HexStr(ssTx.begin(), ssTx.end());
-    CBlockIndex* pindexPrev = NULL;
-    CBlock block;
-    map<uint256, CBlockIndex*>::iterator mi = mapBlockIndex.find(hashBlock);
-    if (mi == mapBlockIndex.end())
+
+    //Deserialize
+    MiningCPID bb = DeserializeBoincBlock(block.vtx[0].hashBoinc,block.nVersion);
+
+    res["height"]=pindex->nHeight;
+    res["version"]=block.nVersion;
+
+    res["cpid"]=bb.cpid;
+    res["Magnitude"]=bb.Magnitude;
+    res["Interest"]=bb.InterestSubsidy;
+    res["BoincReward"]=bb.ResearchSubsidy;
+    res["Difficulty"]=GetBlockDifficulty(block.nBits);
+
+
+    if(fDebug)
     {
-            return ""; //not found
+        res["xbbBoincPublicKey"]=bb.BoincPublicKey;
+
+        res["xbbOrganization"]=bb.Organization;
+        res["xbbClientVersion"]=bb.clientversion;
+
+        res["xbbNeuralHash"]=bb.NeuralHash;
+        res["xbbCurrentNeuralHash"]=bb.CurrentNeuralHash;
+        res["xbbNeuralContractSize"]=bb.superblock.length();
     }
-    pindexPrev = (*mi).second;
-    if (!block.ReadFromDisk(pindexPrev))
+    else
     {
-            return ""; //failed to read
+        if(bb.superblock.length()>=20)
```

Comment:
You could change these to:
`res["IsSuperBlock"] = (block.nIsSuperBlock != 0);`

or:
`res["IsSuperBlock"] = IsSuperBlock(&block);`

Reply:
You mean pblockIndex?
I guess I can, pindex flags should be consistent with block, but i wanted to display the value from boinchash.

Reply:
Ah, I see. You know what, leave that for now. We can add a `IsValidSuperBlock(const std::string&)` and replace all the magic length checks later.